packages/web/components/notifi/notifi-modal-context.tsx

```diff
@@ -0,0 +1,109 @@
+import { useNotifiSubscriptionContext } from "@notifi-network/notifi-react-card";
+import {
+  createContext,
+  FunctionComponent,
+  PropsWithChildren,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+
+import { ModalBaseProps } from "~/modals";
+
+import { useNotifiConfig } from "./notifi-config-context";
+
+type Location = "history" | "expired" | "signup" | "edit" | "historyDetail";
+
+interface NotifiModalFunctions {
+  account: string;
+  innerState: Partial<ModalBaseProps>;
+  isOverLayEnabled: boolean;
+  setIsOverLayEnabled: (isOverLayEnabled: boolean) => void;
+  setLocation: (newLocation: Location) => void;
+}
+
+const NotifiModalContext = createContext<NotifiModalFunctions>({
+  innerState: {},
+} as unknown as NotifiModalFunctions);
+
+export const NotifiModalContextProvider: FunctionComponent<
+  PropsWithChildren<{ account: string }>
+> = ({ account, children }) => {
+  const { setCardView } = useNotifiSubscriptionContext();
+  const [innerState, setInnerState] = useState<Partial<ModalBaseProps>>({});
+  const [location, setLocation] = useState<Location>("signup");
+  const [isOverLayEnabled, setIsOverLayEnabled] = useState(false);
+  const config = useNotifiConfig();
+  const titles = useMemo(() => {
+    if (config.state === "fetched" && config.data.titles?.active) {
+      return config.data.titles;
+    }
+    return undefined;
+  }, [config]);
+
+  useEffect(() => {
```

Comment:
This use of useEffect doesn't seem necessary/idiomatic at first glance, but I wouldn't bother fixing it as long as it works.